libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFAction.m

```diff
@@ -97,12 +97,17 @@ - (NSURL*)urlForActionWithError:(NSError**)error {
         }
         return nil;
     }
-    
-    if ([baseURL.absoluteString hasSuffix:@"/"] && [path hasPrefix:@"/"]) {
-        [path deleteCharactersInRange:NSMakeRange(0, 1)];
+
+    // 'baseUrl' needs to account for URLs with extra trailing slashes, like custom community URLs.
+    NSMutableString *baseUrlString = [NSMutableString stringWithFormat:@"%@", baseURL.absoluteString];
+    if ([baseUrlString hasSuffix:@"/"]) {
+        [baseUrlString deleteCharactersInRange:NSMakeRange(0, 1)];
     }
-    
-    NSURL *url = [NSURL URLWithString:path relativeToURL:baseURL];
+    if (![path hasPrefix:@"/"]) {
+        baseUrlString = [NSMutableString stringWithFormat:@"%@/", baseUrlString];
+    }
+    NSString *urlString = [NSString stringWithFormat:@"%@%@", baseUrlString, path];
```

Comment:
Can't we do that with [pathWithComponents](https://developer.apple.com/reference/foundation/nsstring/1414489-pathcomponents?language=objc)?